### PR TITLE
fix(tx-history): swap "via {provider}" badge uses the swapper brand logo

### LINKED ISSUE #4880: fix(tx-history): swap "via {provider}" badge shows the source-coin icon, not the swapper's brand logo
## Summary

On the Transaction History screen, a completed/in-progress swap card shows a **"via {provider}"** badge (e.g. *via THORChain*). The little icon in that badge currently renders the **source-coin icon**, but it should render the **swapper's brand logo** — the same provider logo we already show on the swap details / route-selection screen.

## Where

- Badge: `VultisigApp/VultisigApp/Features/TransactionHistory/Views/TransactionHistoryCardView.swift` → `viaBadge` (currently `AsyncImageView(logo: transaction.coinLogo, …, ticker: transaction.coinTicker, …)`).
- Correct reference (what it should look like): `VultisigApp/VultisigApp/Features/Swap/Views/SelectRouteSettingsView.swift` uses `AsyncImageView(logo: quote.providerLogo, …)`, where `providerLogo` is defined in `VultisigApp/VultisigApp/Features/Swap/Views/SwapQuote.swift` (maps `SwapProvider` → brand asset name: `THORChain`, `Maya protocol`, `1Inch`, `LI.FI`, `kyberswap`, `swapkit`, `jupiter`).

## Why it's not a one-liner

Transaction History persists the provider only as a **display-name String** (`TransactionHistoryItem.swapProvider`, a SwiftData `@Model` — adding a stored logo field would be a schema migration, avoid). That string varies by record path: `"THORChain"`, `"THORChain-Stagenet"`, `"Maya protocol"` vs `"Maya Protocol"`, `"SwapKit (Chainflip)"`, plus generic provider names. So the fix maps that string → brand asset name with **case-insensitive prefix matching**, mirroring `SwapProvider.providerLogo`. `AsyncImageView` already falls back to a monogram for unknown assets, so the badge's `ticker` should be the provider name (not the coin ticker) to keep the fallback sensible.

## Acceptance

- The "via {provider}" badge icon shows the provider's brand logo (matching the swap route/details screen) for THORChain, Maya, 1inch, KyberSwap, LI.FI, SwapKit, Jupiter.
- Unknown/legacy provider strings fall back to a provider-name monogram, not the coin icon.
- No SwiftData schema change.

### PR DESCRIPTION
**Pull request by gastonm5**
Closes #4880

## Problem
On the Transaction History screen, a swap card's **"via {provider}"** badge rendered the **source-coin** icon instead of the **swapper's brand logo** — unlike the swap route/details screen, which shows the provider's brand.

## Approach: one shared source of truth
Rather than duplicate the brand-asset names, this extracts a single canonical provider identity that both the live swap flow and persisted Transaction History read from, so the two can never drift.

- **New `SwapProviderKind`** (`Blockchain/Swaps/Common/SwapProviderKind.swift`) — a payload-free enum owning:
  - `providerLogo` — the brand imageset name (same strings as before).
  - `displayName` — the canonical brand display name.
  - `init?(persistedName:)` — a tolerant resolver from a **persisted** display string (history stores only the display name, and it varies: `THORChain-Stagenet`, `Maya protocol`/`Maya Protocol`, `SwapKit (Chainflip)`, …), using case-insensitive substring matching.
  - Pure presentation — no Codable/signing/wire involvement.
- **`SwapQuote`** now delegates `providerLogo` → `kind.providerLogo` and `displayName` → `kind.displayName`. The `displayName` output stays **byte-identical** to before (network variants keep their `-Chainnet` / `-Stagenet` suffix), because that string is persisted.
- **`TransactionHistoryData.swapProviderLogo`** resolves via `SwapProviderKind(persistedName:)?.providerLogo` (the bespoke `providerLogoAsset(for:)` map was deleted).
- **`viaBadge`** uses `logo: transaction.swapProviderLogo ?? provider` and `ticker: provider`, so providers without a bundled asset fall back to `AsyncImageView`'s monogram from the provider's initial. Size stays 16×16.

Matching uses the full brand tokens (`thorchain` / `li.fi`), not the loose `thor` / `lifi` substrings (which collide with words like "Author" / "amplifier"), and resolves the outer **SwapKit** brand before its inner protocol for sub-provider strings. No schema change, no new user-facing strings.

## Local gate result
- **Build/Test:** `** TEST SUCCEEDED **` (`XCODEBUILD_EXIT=0`) — full suite, **3505 tests, 1 skipped, 0 failures**, on iPhone 16 sim. Includes `SwapProviderKindTests` and the repointed end-to-end `TransactionHistoryProviderLogoTests`.
- **Lint:** SwiftLint `0 violations` on all changed files.
- **Cross-model review:** one read-only Codex pass over the branch diff. It confirmed the `SwapQuote` display/logo/kind mappings are byte-identical/correct, and flagged one Medium issue — a loose `thor` marker false-matching "Author" and mis-ordering a `SwapKit (THORChain)` sub-provider. **Fixed** (full-token match + swapkit-first ordering) with regression tests; re-ran green.

## Test coverage
- `SwapProviderKindTests` — `providerLogo` + `displayName` per case; `init(persistedName:)` for every variant/casing/whitespace/sub-provider suffix; nil for empty/unknown; and the `Author`/`amplifier` false-match + `SwapKit (THORChain)` ordering regressions.
- `TransactionHistoryProviderLogoTests` — end-to-end `swapProviderLogo` for every documented persisted string → expected asset, and nil/empty/unknown → nil.

## Summary by CodeRabbit

* **New Features**
  * Added branded swap-provider logos to transaction history “via {provider}” badges.
  * Introduced a unified swap-provider identity to keep provider naming and branding consistent across swap quotes and transaction history.
* **Bug Fixes**
  * Improved mapping from persisted provider strings to the correct brand, including trimming, case-insensitive matching, and safer token-boundary detection.
  * Updated badge rendering to use swapper branding when available (and fall back appropriately when not).
* **Tests**
  * Added new test suites covering provider logo/name resolution and persisted-name parsing, including network variants and edge cases.

### CODE REVIEW

File: VultisigApp/VultisigApp/Blockchain/Swaps/Common/SwapProviderKind.swift [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        case .kyberSwap:
+            return "KyberSwap"
+        case .lifi:
+            return "LI.FI"
+        case .swapkit:
+            return "SwapKit"
+        case .jupiter:
+            return "Jupiter"
+        }
+    }
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/Swaps/Common/SwapProviderKind.swift:66]: Declining this one — keeping provider names unlocalized here is intentional:

- **They're brand proper nouns** (THORChain, Maya, 1Inch, KyberSwap, LI.FI, SwapKit, Jupiter). Per this repo's localization guidance, crypto/brand terms stay in English across all locales — we don't translate "THORChain".
- **`providerLogo` returns Assets-catalog imageset names**, not user-facing copy — localizing those would break the image lookup outright.
- **`displayName` is persisted** into Transaction History (`swapProvider`) and then used as a *locale-independent matching key* (`SwapProviderKind(persistedName:)`) and as an explorer-link input. Localizing it would make stored data locale-dependent and could break provider resolution / explorer links for existing rows.
- These literals also **pre-date this PR** — the refactor only relocated them byte-identically into the shared `SwapProviderKind`.

Resolving as won't-fix.

File: VultisigApp/VultisigApp/Features/TransactionHistory/Models/TransactionHistoryData.swift [hunk 2]
```diff
@@ -184,6 +184,46 @@ extension TransactionHistoryData {
...
+            return nil
+        }
+
+        if normalized.contains("thorchain") { return "THORChain" }
+        if normalized.contains("maya") { return "Maya protocol" }
+        if normalized.contains("1inch") { return "1Inch" }
+        if normalized.contains("kyber") { return "kyberswap" }
+        if normalized.contains("li.fi") { return "LI.FI" }
+        if normalized.contains("swapkit") { return "swapkit" }
+        if normalized.contains("jupiter") { return "jupiter" }
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Features/TransactionHistory/Models/TransactionHistoryData.swift]: Follow-up: took this further in `344a616a3` — swapped the loose `contains` for prefix + non-alphanumeric-boundary matching in `SwapProviderKind.init?(persistedName:)`. The exact near-matches flagged here (`SwapKitty`, `not-thorchain`) now fall through to `nil` → the provider-name monogram fallback, with regression tests covering them (plus `mayachain`, `Author`, `amplifier`). All documented variants — including sub-provider forms like `SwapKit (THORChain)` — still resolve. Full suite green (3506 tests, 0 failures).

**Review by johnnyluo — APPROVED** [review]
LGTM
